couchdb-cluster-admin/file_plan.py

```diff
@@ -3,40 +3,42 @@
 import json
 from utils import get_config_from_args, get_shard_allocation, set_up_parser
 from describe import print_shard_table
+from doc_models import ShardAllocationDoc
 
 
 def read_plan_file(filename):
     with open(filename) as f:
-        return json.load(f)
+        plan = json.load(f)
 
+    return {
+        db_name: ShardAllocationDoc.from_plan_json(db_name, plan_json)
+        for db_name, plan_json in plan.items()
+    }
 
 def update_shard_allocation_docs_from_plan(shard_allocation_docs, plan):
     shard_allocation_docs = {shard_allocation_doc.db_name: shard_allocation_doc
                              for shard_allocation_doc in shard_allocation_docs}
-    for db_name, by_range in sorted(plan.items()):
+    for db_name, allocation_doc in sorted(plan.items()):
         shard_allocation_doc = shard_allocation_docs[db_name]
-        by_node = defaultdict(list)
-        for shard, nodes in by_range.items():
-            for node in nodes:
-                by_node[node].append(shard)
-        shard_allocation_doc.by_range = by_range
-        shard_allocation_doc.by_node = by_node
+        shard_allocation_doc.by_range = allocation_doc.by_range
```

Comment:
may want a better contrast than `shard_allocation_doc` for the one coming from the cluster vs `allocation_doc` the one coming from the plan. Maybe (if I got the semantic concept right) `cluster_allocation_doc` vs `plan_allocation_doc`? (I also realize I kind of started it with some inconsistent naming throughout.)